crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -498,8 +477,15 @@ pub(crate) fn check_docstring(
     };
 
     // DOC201
-    if checker.enabled(Rule::DocstringMissingReturns) {
-        if !is_property(definition, checker) && docstring_sections.returns.is_none() {
+    if checker.enabled(Rule::DocstringMissingReturns) && docstring_sections.returns.is_none() {
+        let extra_property_decorators = checker
+            .settings
+            .pydocstyle
+            .property_decorators
+            .iter()
+            .map(|decorator| QualifiedName::from_dotted_name(decorator))
+            .collect::<Vec<QualifiedName>>();
```

Comment:
I think I'm reviewing the PRs in the wrong order. This is now behind your new iterator and no longer requires collecting?

Reply:
> I think I'm reviewing the PRs in the wrong order.

You are, yes :-) This is the first PR in the stack, and the other two PRs are based on top of this branch.

But as I mentioned in https://github.com/astral-sh/ruff/pull/12582#discussion_r1697209176, I think we still need to collect ~always even with my new iterator, because the common pattern seems to be to do something like this:

```rs
for decorator in decorators {
    if extra_properties.any(|property| property == decorator) {
        return true;
    }
}
```

If `extra_properties` is an iterator there, it could be exhausted after the first iteration of the outer `for` loop

Reply:
We can `Clone` the iterator to avoid this

Reply:
Oh, good point.